pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2045,20 +2045,20 @@ private void handlePoliciesUpdates(NamespaceName namespace) {
                         if (namespace.includes(TopicName.get(name))) {
                             // If the topic is already created, immediately apply the updated policies, otherwise
                             // once the topic is created it'll apply the policies update
-                            topicFuture.thenAccept(topic -> {
+                            topicFuture.thenAcceptAsync(topic -> {
```

Comment:
The context thread is changed by line 2036, maybe we don't need to change it again.